test/thermo/PengRobinsonMFTP_Test.cpp

```diff
@@ -0,0 +1,199 @@
+    }
+}
+
+TEST_F(PengRobinsonMFTP_Test, standardConcentrations)
+{
+    EXPECT_DOUBLE_EQ(test_phase->pressure()/(test_phase->temperature()*GasConstant), test_phase->standardConcentration(0));
+    EXPECT_DOUBLE_EQ(test_phase->pressure()/(test_phase->temperature()*GasConstant), test_phase->standardConcentration(1));
+}
+
+TEST_F(PengRobinsonMFTP_Test, activityConcentrations)
+{
+    // Check to make sure activityConcentration_i == standardConcentration_i * gamma_i * X_i
+    vector_fp standardConcs(7);
+    vector_fp activityCoeffs(7);
+    vector_fp activityConcentrations(7);
+    double xmin = 0.6;
+    double xmax = 0.9;
+    int numSteps = 9;
+    double dx = (xmax-xmin)/(numSteps-1);
+
+    for(int i=0; i < 9; ++i)
+    {
+        const double r = xmin + i*dx;
+        set_r(r);
+        test_phase->getActivityCoefficients(&activityCoeffs[0]);
+        standardConcs[0] = test_phase->standardConcentration(0);
+        standardConcs[2] = test_phase->standardConcentration(2);
+        test_phase->getActivityConcentrations(&activityConcentrations[0]);
+
+        EXPECT_NEAR(standardConcs[0] * r * activityCoeffs[0], activityConcentrations[0], 1.e-6);
+        EXPECT_NEAR(standardConcs[2] * (1-r) * activityCoeffs[2], activityConcentrations[2], 1.e-6);
+    }
+}
+
+TEST_F(PengRobinsonMFTP_Test, setTP)
+{
+    // Check to make sure that the phase diagram is accurately reproduced for a few select isobars
+
+    // All sub-cooled liquid:
+    const double p1[6] = {
+		1.666169531425807e+002,
+		1.610359415729891e+002,
+		1.561558755014150e+002,
+		1.518163520394436e+002,
+		1.479077426738997e+002,
+		1.443514857504623e+002		
+    };
+    // Phase change between temperatures 4 & 5:
+    const double p2[6] = {
+		7.085417135252475e+002,
+		6.715912546208409e+002,
+		6.252218774469104e+002,
+		5.555249258326570e+002,
+		3.218596258615833e+002,
+		2.791272952614171e+002
+    };
+    // Supercritical; no discontinuity in rho values:
+    const double p3[6] = {
+		7.695965571714201e+002,
+		7.455746209641901e+002,
+		7.196124816622006e+002,
+		6.912592525839008e+002,
+		6.598814961519255e+002,
+		6.245690038776825e+002
+    };
+
+    for(int i=0; i<6; ++i)
+    {
+        const double temp = 294 + i*2;
+        set_r(0.99);
+        test_phase->setState_TP(temp, 5542027.5);
+        EXPECT_NEAR(test_phase->density(),p1[i],1.e-8);
+
+        test_phase->setState_TP(temp, 7389370.);
+        EXPECT_NEAR(test_phase->density(),p2[i],1.e-8);
+
+        test_phase->setState_TP(temp, 9236712.5);
+        EXPECT_NEAR(test_phase->density(),p3[i],1.e-8);
+    }
+}
+
+TEST_F(PengRobinsonMFTP_Test, getPressure)
+{
+	// Check to make sure that the P-R equation is accurately reproduced for a few selected values
+
+	double a_coeff = 3.958134E+5;
+	double b_coeff = 26.6275/1000;
+	double acc_factor = 0.228;
+	double pres_theoretical, kappa, alpha, mv;
+	const double Tcrit = test_phase->critTemperature();
+
+	//Calculate kappa value
```

Comment:
Can you provide a brief comment here to explain the hard-coded coefficients in the equations below?  Briefly, what are they based on?  Do they vary as a function of species or composition?

Reply:
These hard-coded values are correlated with acentric factor data and come from the original paper by Peng and Robinson. More details are given in the main documentation file.